### PR TITLE
Add Fedora 35 and 36 to the test suite

### PR DESCRIPTION
**Pull request by yubiuser**
**What does this PR aim to accomplish?:**

Currently, we support Fedora 34. However, this is EOL since 2022-06-07 (https://docs.fedoraproject.org/en-US/releases/eol/). Supported release are Fedora  35 and 36, which we should support as well.

Fixes: https://github.com/pi-hole/pi-hole/issues/4796

- **What documentation changes (if any) are needed to support this PR?:**

Documentation at: https://github.com/pi-hole/docs/pull/781

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

**Comment by yubiuser** [comment]
We could also drop support/tests for Fedora 34 if you agree.

**Comment by yubiuser** [comment]
Will need changes to `versions.pi-hole.net` after release.

**Comment by dschaper** [comment]
> We could also drop support/tests for Fedora 34 if you agree.

I've requested a review from Canada but I don't see any issues with dropping 34. Is there anything that requires 35+ and will prevent 34 from working or is it a housekeeping task to clean it out?

**Comment by yubiuser** [comment]
Houskeeping. But currently the test for 35/36 are failing. I investigate. Maybe a change in tests/installer prevents 34 afterwards.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -48,7 +48,16 @@ jobs:
...
       matrix:
         distro:
-          [debian_10, debian_11, ubuntu_20, ubuntu_22, centos_8, fedora_34]
+          [
+            debian_10,
+            debian_11,
+            ubuntu_20,
+            ubuntu_22,
+            centos_8,
+            fedora_34,
```
Comment (bcambl) [inline .github/workflows/test.yml:57]: `fedora_34` can be removed

**Comment by bcambl** [comment]
All _should_ be good to bump up the supported versions.

**Comment by yubiuser** [comment]
I just tried the installer in a Fedora 35 VM and it worked without issues....

**Comment by yubiuser** [comment]
I figured it out - it's a catch22. 
The test are failing because the Fedora 35 & 36 docker images lacking `initscripts`. But we need this here

https://github.com/pi-hole/pi-hole/blob/d88629e23fcfbf52b97bf14800bfc510b0284be6/pihole#L142-L150

to run `service pihole-FTL restart`. This will create `/etc/pihole/dhcp.leases` for which we check at 
https://github.com/pi-hole/pi-hole/blob/d88629e23fcfbf52b97bf14800bfc510b0284be6/test/test_any_automated_install.py#L231

The file is created during service startup: https://github.com/pi-hole/pi-hole/blob/d88629e23fcfbf52b97bf14800bfc510b0284be6/advanced/Templates/pihole-FTL.service#L37)

___
Fedora uses `systemd` and we could just replace `service` with `systemctl` to start the service. But we mock the service 
https://github.com/pi-hole/pi-hole/blob/d88629e23fcfbf52b97bf14800bfc510b0284be6/test/test_any_automated_install.py#L160-L180

and, more importantly, we test **after** we have downloaded and installed `pihole` from the current `master` branch which would still contain the `service`command.

___

I see three options here: 

1) create another PR which will replace the `service pihole-FTL restart` with `systemctl restart pihole-FTL` and merge this up to master. Afterwards this PR should not fail anymore
2) add `initscripts` as a dependency
3) switch to another Fedora image

**Comment by yubiuser** [comment]
I tried for two days, but

> create another PR which will replace the service pihole-FTL restart with systemctl restart pihole-FTL and merge this up to master. Afterwards this PR should not fail anymore

could not get this working.  There is no initsystem present in the container, and I could not get `systemd` to run properly on the container. It just did not start the `pihole-FTL.service`.
___

I'll therefore go for

> add initscripts as a dependency

Any initsystem should present in a "real" Fedora system.

**Review by dschaper — APPROVED** [review]

**Pull request merged**